platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/Capture.kt

```diff
@@ -220,21 +241,155 @@ object Capture {
+        ) {
+            startAsync(
+                apiKey,
+                sessionStrategy,
+                configuration,
+                completionResult,
+                fieldProviders,
+                dateProvider,
+                apiUrl,
+                CaptureJniLibrary,
+            )
+        }
+
+        @Synchronized
+        @JvmStatic
+        @JvmOverloads
+        internal fun startAsync(
+            apiKey: String,
+            sessionStrategy: SessionStrategy,
+            configuration: Configuration = Configuration(),
+            completionResult: (CaptureResult<StartedSdk>) -> Unit,
+            fieldProviders: List<FieldProvider> = listOf(),
+            dateProvider: DateProvider? = null,
+            apiUrl: HttpUrl = defaultCaptureApiUrl,
+            bridge: IBridge,
+            backgroundThreadHandler: IBackgroundThreadHandler = CaptureDispatchers.CommonBackground,
+        ) {
+            // Note that we need to use @Synchronized to prevent multiple loggers from being initialized,
+            // while subsequent logger access relies on volatile reads.
+
+            // Ideally we would use `getAndUpdate` in here but it's available for API 24 and up only.
+            if (default.compareAndSet(
+                    LoggerState.NotStarted,
+                    LoggerState.Starting,
+                )
+            ) {
+                backgroundThreadHandler.runAsync {
+                    // There's nothing we can do if we don't have yet access to the application context.
+                    if (!ContextHolder.isInitialized) {
+                        Log.w(
+                            LOG_TAG,
+                            "Attempted to initialize Capture with a null context",
+                        )
+                        completionResult.invoke(CaptureResult.Failure(StartAsyncWithNullContext))
+                        return@runAsync
+                    }
+
+                    try {
+                        preInitInMemoryLoggerRef.compareAndSet(null, PreInitInMemoryLogger())
+                        if (configuration.enableFatalIssueReporting) {
+                            fatalIssueReporter.initBuiltInMode(ContextHolder.APP_CONTEXT)
+                        }
+                        val loggerImpl =
+                            LoggerImpl(
+                                apiKey = apiKey,
+                                apiUrl = apiUrl,
+                                fieldProviders = fieldProviders,
+                                dateProvider = dateProvider ?: SystemDateProvider(),
+                                configuration = configuration,
+                                sessionStrategy = sessionStrategy,
+                                bridge = bridge,
+                                fatalIssueReporter = fatalIssueReporter,
+                                preInitLogFlusher = preInitInMemoryLoggerRef.get(),
+                            )
```

Comment:
Inside LoggerImpl's init method, it calls the flusher to flush the pre-init logger, but we don't clear `preInitInMemoryLoggerRef` until the finally clause. Wouldn't that leave a hole for another thread to get the pre-init log after the flush, but before switching the instance to the regular logger (effectively losing that log)?

What if we didn't pass the flusher into the logger, but instead built the logger fully out here, did an atomic switch of the reference, and then finally did a flush of the now outside-inaccessible pre-init logger?